helix-core/src/main/java/org/apache/helix/controller/stages/CurrentStateComputationStage.java

```diff
@@ -69,6 +78,13 @@ public void process(ClusterEvent event) throws Exception {
       updateCurrentStates(instance, currentStateMap.values(), currentStateOutput, resourceMap);
     }
     event.addAttribute(AttributeName.CURRENT_STATE.name(), currentStateOutput);
+
+    if (cache instanceof ResourceControllerDataProvider) {
+      final ClusterStatusMonitor clusterStatusMonitor =
```

Comment:
I might be wrong, but I remember this object could be null. Please add a check here.